### PR TITLE
Use kernel.project_dir if available to support symfony 5

### PR DESCRIPTION
**Pull request by acrobat**
| Q             | A
| ------------- | ---
| Branch?       | "master" for new features / the branch of the current release for fixes
| Bug fix?      | yes
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | 
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/HttpKernel/TestKernel.php [hunk 1]
```diff
@@ -166,6 +166,15 @@ public function getKernelDir()
         return $kernelDir;
     }
 
+    public function getProjectDir()
```
Comment (dbu) [inline src/HttpKernel/TestKernel.php:169]: as this is the same as getKernelDir, should we make this one call getKernelDir? then a using bundle overwriting getKernelDir would also work correctly.

File: resources/config/default.php [hunk 2]
```diff
@@ -23,7 +23,7 @@
     }
 }
 
-$loader->import('dist/parameters.yml');
+$loader->import('dist/parameters_sf5.yml');
```
Comment (dbu) [inline resources/config/default.php:26]: can we just fixed go to the sf5 version? or do we need an if/else here?

File: src/HttpKernel/TestKernel.php (hunk 1 above)
Comment (alexander-schranz) [reply src/HttpKernel/TestKernel.php:169]: I think we need to overwrite the `getProjectDir` else `getProjectDir` will search the folders up to the `composer.json` and thats mostly not what you want in a TestKernel.

Depends on the bc we remove `getKernelDir` or just change it to:

```php
public function getProjectDir()
{
    return $this->getKernelDir();
}
```

File: resources/config/default.php (hunk 2 above)
Comment (acrobat) [reply resources/config/default.php]: Yes indeed, it seems I forgot to add the if/else. Should be fixed now!

File: src/HttpKernel/TestKernel.php (hunk 1 above)
Comment (acrobat) [reply src/HttpKernel/TestKernel.php:165]: I didn't the otherway around, as kernel rootdir will be removed in the future when you drop sf <4.x support. So I've called `getProjectDir` in `getKernelDir`. This way we can easily cleanup the old method in the future.

Is it ok like this or should I switch it around?

Comment (dbu) [reply src/HttpKernel/TestKernel.php:165]: that makes sense to me. its "only" testing, so if a project had to overwrite getKernelDir, they will now notice when they upgrade and have to change their code to overwrite getProjectDir instead.

Comment (alexander-schranz) [reply src/HttpKernel/TestKernel.php:165]: 👍 looks good for me!

**Pull request merged**